### ISSUE TITLE
Why does discriminator's output change between batchsize:64 and batchsize:1 on inference.

### ISSUE THREAD

**Issue by y-shirai-r** [pytorch/examples#514]
I'm trying to get the discriminator output using train finished discriminator.
Procedure is below.
1. training dcgan.
2. preparing my image data and resize it 64 * 64.
3. load my image data using dataloader(same as training's one.)
4. I change only batch size at inference.
5. I got small Discriminator outputs(after sigmoid result).

for example)
I prepared 64 images under the dataroot directory. And I tried 2 experience.

I got a below's discriminator output at inference using batch size 64.
```
tensor([0.9955, 0.8801, 0.9727, 0.7377, 0.2667, 0.9432, 0.9941, 0.6896, 0.8638,
        0.5006, 0.9766, 0.4148, 0.9577, 0.9065, 0.9849, 0.9027, 0.1619, 0.5418,
        0.9256, 0.7502, 0.1467, 0.8197, 0.9100, 0.3416, 0.0066, 0.9521, 0.9973,
        1.0000, 0.4952, 0.3026, 0.5347, 0.8695, 0.8033, 0.6709, 0.3602, 0.2145,
        0.6901, 0.0129, 0.6780, 0.5321, 0.8195, 0.8662, 0.1759, 0.5599, 0.7313,
        0.5138, 0.9396, 0.9256, 0.3011, 0.8163, 0.8046, 0.4802, 0.6256, 0.1656,
        0.9368, 0.1080, 0.5960, 0.9493, 0.9533, 0.9609, 0.0137, 0.1603, 0.7717,
        0.5684], device='cuda:0', grad_fn=<SqueezeBackward1>)
```

I got a below's discriminator output at inference using batch size 1.
```
tensor([0.6289], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.2455], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.8702], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.0000], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.0002], device='cuda:0', grad_fn=<SqueezeBackward1>)
............
tensor([0.0002], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.0022], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.9955], device='cuda:0', grad_fn=<SqueezeBackward1>)
tensor([0.1370], device='cuda:0', grad_fn=<SqueezeBackward1>)
```

I wonder why I got a small output(different from batch size64) on batch size1. 
For instance, I got the 0.0000 on batch size 1, but at batch size 64 0.0000 is nothing.

Also I tried to match tensor size using torch.cat at batch size 1.
I changed from [1, 3, 64, 64] -> [64, 3, 64, 64], using same one image's tensor and torch.cat.
But I got different output value.

If you have any suggestions or point out then please let me know.

**Comment by soumith** [comment]
model is in training mode, and there is BatchNorm.

before you do inference, do `model.eval()`

**Comment by y-shirai-r** [comment]
Thank you so much.
I can confirm matching between batch_size:64 and batch_size:1.
